crates/rover-client/src/shared/check_response.rs

```diff
@@ -10,42 +10,80 @@ use serde::{Deserialize, Serialize};
 use prettytable::{row, Table};
 use serde_json::{json, Value};
 
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub enum CheckResponse {
+    OpeartionCheckResponse(OperationCheckResponse),
+    OpeartionLessCheckResponse(OpeartionLessCheckResponse),
+}
+
+impl CheckResponse {
+    pub fn get_json(&self) -> Value {
+        match self {
+            CheckResponse::OpeartionCheckResponse(operation_check_response) => {
+                operation_check_response.get_json()
+            }
+            CheckResponse::OpeartionLessCheckResponse(operation_less_check_response) => {
+                operation_less_check_response.get_json()
+            }
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub struct OpeartionLessCheckResponse {
+    pub target_url: Option<String>,
+    pub core_schema_modified: bool,
+    pub contains_operation_check: bool,
+}
+
+impl OpeartionLessCheckResponse {
+    pub fn to_output(&self) -> String {
+        "".to_string()
+    }
+
+    pub fn get_json(&self) -> Value {
+        json!(self)
+    }
+}
+
 /// CheckResponse is the return type of the
 /// `graph` and `subgraph` check operations
 #[derive(Debug, Serialize, Clone, Eq, PartialEq)]
-pub struct CheckResponse {
+pub struct OperationCheckResponse {
     target_url: Option<String>,
     operation_check_count: u64,
     changes: Vec<SchemaChange>,
     #[serde(skip_serializing)]
```

Comment:
i wonder why we chose to skip serializing this. do you think this would be valuable to include in the check json output?

Reply:
Yea. I definitely think it would be useful. I guess it is mirrored by the exit code, so that might be why?
I'm gonna remove the macro.

Reply:
I added it to the codebase.